### PR TITLE
CCP-708: Send parent uid to Econda, new config setting 'sendParentUid'

### PR DESCRIPTION
**Pull request by patrick-scherer**

### CODE REVIEW

File: frontend/helpers.js [hunk 1]
```diff
@@ -31,16 +31,18 @@ export const getEmosproForUid = uid => ({
  * Returns the data for the ec_Event of a product or variant.
  * @param {Object} [baseProduct] The base (parent) product.
  * @param {Object} [variant] The selected variant.
+ * @param {boolean} sendParentUid The parent uid will be sent. Even for a variant article.
```
Comment (Carsten89) [inline frontend/helpers.js]: jsDoc: please mark this param as optional and add the default value

File: frontend/helpers.js [hunk 2]
```diff
@@ -31,16 +31,18 @@ export const getEmosproForUid = uid => ({
...
   const product = variantAvailable ? variant : baseProduct;
+  const productUid = (sendParentUid && variantAvailable) ? baseProduct.uid : product.uid;
 
   return {
     type: 'view',
     pid: baseProduct ? baseProduct.uid : '',
-    sku: variant.uid ? variant.uid : '',
+    sku: productUid,
     name: product.name,
     price: parseFloat(product.amount.net),
```
Comment (Carsten89) [inline frontend/helpers.js]: i guess it should also take the price from the parent if `sendParentUid` is true?

File: frontend/helpers.js [hunk 3]
```diff
@@ -31,16 +31,18 @@ export const getEmosproForUid = uid => ({
...
   const variantAvailable = !!Object.keys(variant).length;
   const product = variantAvailable ? variant : baseProduct;
+  const productUid = (sendParentUid && variantAvailable) ? baseProduct.uid : product.uid;
 
   return {
     type: 'view',
     pid: baseProduct ? baseProduct.uid : '',
-    sku: variant.uid ? variant.uid : '',
+    sku: productUid,
     name: product.name,
```
Comment (Carsten89) [inline frontend/helpers.js]: same as for the price. check other comment

File: frontend/helpers.js [hunk 4]
```diff
@@ -31,16 +31,18 @@ export const getEmosproForUid = uid => ({
...
  * @param {Object} [baseProduct] The base (parent) product.
  * @param {Object} [variant] The selected variant.
+ * @param {boolean} sendParentUid The parent uid will be sent. Even for a variant article.
  * @return {Object} ec_Event data of the given product
  */
-export const getProductEventData = (baseProduct = {}, variant = {}) => {
+export const getProductEventData = (baseProduct = {}, variant = {}, sendParentUid = false) => {
   const variantAvailable = !!Object.keys(variant).length;
   const product = variantAvailable ? variant : baseProduct;
+  const productUid = (sendParentUid && variantAvailable) ? baseProduct.uid : product.uid;
```
Comment (Carsten89) [inline frontend/helpers.js]: why do you check here if `variantAvailable` is true?

File: README.md [hunk 5]
```diff
@@ -4,15 +4,16 @@ Implementation of the default econda tracking plus ecommerce tracking.
...
 
 **Example Config:** 
 ```
 {
     "siteid": "de",
     "clientKey": "1234-acbd-",
     "langid": "de",
-    "countryid": "de"
+    "countryid": "de",
+    "sendParentUid": false
```
Comment (Carsten89) [inline README.md]: please add the new config option also here https://wiki.shopgate.guru/display/IN/How+to+use+Shop+Configs

**Review by Carsten89 — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (patrick-scherer) [reply README.md]: done

File: frontend/helpers.js (hunk 1 above)
Comment (patrick-scherer) [reply frontend/helpers.js]: done

File: README.md [hunk 6]
```diff
@@ -4,15 +4,16 @@ Implementation of the default econda tracking plus ecommerce tracking.
 
 ## Configuration
 
-The extension get the values for clientKey, siteid, langid and countryid from a shop config (Key: `sg_connect_tracking_econda`).
+The extension get the values for clientKey, siteid, langid, countryid and sendParentUid from a shop config (Key: `sg_connect_tracking_econda`).
```
Comment (Carsten89) [inline README.md]: it's `sendParentData` now

File: README.md (hunk 5 above)
Comment (Carsten89) [inline README.md]: it's `sendParentData` now

File: frontend/helpers.js [hunk 7]
```diff
@@ -31,16 +31,26 @@ export const getEmosproForUid = uid => ({
...
+  
+  let product = {};
+
+  if (sendParentData) {
+    product = baseProduct;
+  } else {
+    product = variantAvailable ? variant : baseProduct;
+  }
+
+  console.log(product);
```
Comment (Carsten89) [inline frontend/helpers.js]: no logs please ;)

File: frontend/helpers.js [hunk 8]
```diff
@@ -31,16 +31,26 @@ export const getEmosproForUid = uid => ({
...
  */
-export const getProductEventData = (baseProduct = {}, variant = {}) => {
+export const getProductEventData = (baseProduct = {}, variant = {}, sendParentData = false) => {
   const variantAvailable = !!Object.keys(variant).length;
-  const product = variantAvailable ? variant : baseProduct;
+  
+  let product = {};
+
+  if (sendParentData) {
+    product = baseProduct;
```
Comment (Carsten89) [inline frontend/helpers.js]: If the addToCart is called from outside the PDP, baseProduct is not always available. i added some check

**Review by Carsten89 — CHANGES_REQUESTED** [review]

**Review by Carsten89 — APPROVED** [review]

**Pull request merged**